c10/cuda/CUDAStream.cpp

```diff
@@ -305,9 +323,19 @@ CUDAStream CUDAStream_fromInternals(const LeakyStreamInternals* ptr) {
 } // anonymous namespace
 
 cudaStream_t CUDAStream::stream() const {
-  auto ptr = CUDAStream_internals(*this);
-  AT_ASSERT(ptr);
-  return ptr->stream;
+  int64_t stream_id = unwrap().id();
+  // the stream_ids managed from the pool have only 8 bits
+  // so any value higher than that can be reinterpreted
+  // as an externally allocated stream
+  if (streamIdType(stream_id) == StreamIdType::EXT) {
+    // In this case this is a externally allocated stream
+    // we don't need to manage its life cycle
+    return reinterpret_cast<cudaStream_t>(stream_id);
+  } else {
+    auto ptr = CUDAStream_internals(*this);
+    AT_ASSERT(ptr);
```

Comment:
TORCH_CUDA_ASSERT

Reply:
changed the AT_ASSERTS in the file to TORCH_CUDA_ASSERT

Reply:
```
error: ‘TORCH_CUDA_ASSERT’ was not declared in this scope
   TORCH_CUDA_ASSERT(ptr);
```

actually, there is no `TORCH_CUDA_ASSERT` in the codebase 😅 I grepped it, but didn't find any :(

Reply:
Sorry, TORCH_INTERNAL_ASSERT. AT_ASSERT is deprecated